source/Cosmos.System2/Graphics/VBECanvas.cs

```diff
@@ -336,13 +336,12 @@ private int GetPointOffset(int aX, int aY)
         /// <param name="aHeight">Height.</param>
         public override void DrawFilledRectangle(Pen aPen, int aX, int aY, int aWidth, int aHeight)
         {
-            int xOffset = GetPointOffset(aX, aY);
-            int xScreenWidthInPixel = Mode.Columns * ((int)Mode.ColorDepth / 8);
-            aWidth *= (int)Mode.ColorDepth / 8;
+            //no body knows how any why but we must divide aWidth by 32, not by 8, because otherwise everything would end up 4 times the width
+            aWidth = Math.Min(aWidth, Mode.Columns - aX) * (int)Mode.ColorDepth / 32;
```

Comment:
The reason seems to be that ClearVRAM clears one uint at a time. So we clear pixelwise not byte wise.